src/software/visualizer/drawing/geom.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <QtCore/QLine>
+#include <QtCore/QPoint>
+#include <QtCore/QRect>
+#include <QtGui/QPolygonF>
+#include <QtWidgets/QGraphicsScene>
+
+#include "software/new_geom/circle.h"
+#include "software/new_geom/convex_polygon.h"
+#include "software/new_geom/rectangle.h"
+#include "software/new_geom/segment.h"
+#include "software/visualizer/geom/geometry_conversion.h"
+
+/**
+ * Draws the Rectangle on the given scene.
+ *
+ * @param scene The scene to draw on
+ * @param rectangle The Rectangle to draw
+ * @param pen The QPen to draw the Rectangle
+ * @param brush_opt The optional pointer to a brush to fill the Rectangle
```

Comment:
If you want the interface to be "if no brush is given, the shape has no fill / has transparent fill", you _might_ need to explicitly set a transparent brush.

IIRC Qt might just use the previously set brush when drawing without specifying one, so you should double check that so we don't have previous brushes conflicting with future draw calls.

Reply:
https://doc.qt.io/qt-5/qgraphicsscene.html#addRect iimn it default constructs and brush and pen if they aren't specified